cognite/powerops/custom_modules/power_model_v1/data_models/containers/DateSpecification.container.yaml

```diff
@@ -0,0 +1,49 @@
+space: "{{power_ops_models}}"
+externalId: DateSpecification
+name: DateSpecification
+usedFor: node
+properties:
+  name:
+    type:
+      list: false
+      collation: ucs_basic
+      type: text
+    nullable: false
+    autoIncrement: false
+    name: name
+  processingTimezone:
+    type:
+      list: false
+      collation: ucs_basic
+      type: text
+    nullable: true
+    defaultValue: "UTC"
+    autoIncrement: false
+    name: processingTimezone
+  resultingTimezone:
+    type:
+      list: false
+      collation: ucs_basic
+      type: text
+    nullable: true
+    defaultValue: "UTC"
+    autoIncrement: false
+    name: resultingTimezone
+  floorFrame:
+    type:
+      list: false
+      collation: ucs_basic
+      type: text
+    nullable: true
+    defaultValue: "day"
+    autoIncrement: false
+    name: floorFrame
+    description: The property to "floor" the time to (e.g. "week", "day", "hour", "minute")
+  shiftDefinition:
+    type:
+      list: false
+      type: json
+    nullable: true
+    autoIncrement: false
+    name: shiftDefinition
+    description: "A definition of what time shift to apply to a date (e.g. {'days': 1, 'hours': 2})"
```

Comment:
As I said earlier, I don't think this is flexible enough. I don't think this would allow us to cover all of these examples (only one of A or B, and maybe C but not D):
````
import arrow
input_time = arrow.get("2024-05-26T09:00:00+02:00")

print(input_time.floor("week").shift(days=1)) # (A) Result: 2024-05-21T00:00:00+02:00
print(input_time.shift(days=1).floor("week")) # (B) Result: 2024-05-27T00:00:00+02:00

print(input_time.shift(days=3).floor("week"))               # (C) Result: 2024-05-27T00:00:00+02:00
print(input_time.shift(days=1).floor("week").shift(days=2)) # (D) Result: 2024-05-29T00:00:00+02:00
```

Reply:
I don't think we want to support all of those things because in my opinion half of those don't provide a consistent behavior.

(A) will **always** give you the Tuesday of the "current" week given a Mon-Sun week
(B) will **either** give you the "current" week or next weeks Monday depending on what the current day is (if it's a Sunday you get next Monday if it's Monday-Saturday you get this Monday)
(C) will **either** give you the current Monday or next week's Monday depending on what day of the week you land on in 3 days
(D) will **either** give you the current Wednesday or next week's Wednesday depending on what day of the week is in 3 days

We shouldn't allow for that kind of ambiguity which is too easy to mess up right now with the current implementation. I can't see a use case in which we want an **either** situation depending on what day of the week it is?

Reply:
We should only allow for non-ambiguous statements like these for example:

- this week's Monday
- yesterday
- 2 days from now
- 2 weeks from now
- Wednesday 2 weeks from now
- a month ago from today
- Thursday last week
....

It's easy to ensure this non-ambiguity by _always_ doing floor and then shift

Reply:
No, I disagree. It should be possible to specify something like "The start of the week that tomorrow is in" (shift 1 day, then floor week)

Reply:
And it's not ambiguity, it's just a specification for different things depending on which day of the week we are in (so is `floor("week")` - it will shift by 4.5 days from Friday noon, or by 0.25 days from Monday 06:00...)

Reply:
We are not building a product for only the configurations we have now.

Reply:
I see the trade-off between easy to understand and flexible, but I want the flexibility here. If make a type that support this flexibility, we can make it easy to understand by including some instances with names (or external IDs?) like `tomorrow`, `end_of_next_week` etc. (which we can create as a standard part of resync or something).

Reply:
Resolved from the conversation that `shift(weekday=0)` accounts for what is needed in terms of flexibility